### PR TITLE
LFDNoise: fine tunning small details

### PR DESCRIPTION
**Pull request by tiagmoraismorgado**
greetings everyone

hope you are all doing fine, going throughout a creative mood in this early autumn. I would like to suggest a couple of edits to LFDNoise documentation.  with this contribution, i aim at helping to improve some small details on grammar, syntax, semmantics, and overall language. i am not one hundred percent sure my contribution will be valid. in fact i am not a native english speaker. but at least i used like gitbook to help me to proof read the text. i absolutely accept it if my contribution isn't relevant. feel free to drop any hard criticism. i will be opened to hear, and try to pay attention to the requested details. i leave this entry opened so that anyone can edit it, and have like a meaningful contribution. let's do this as a team. lookng forward to your feedback, made with love. peace. tiago

### CODE REVIEW

File: HelpSource/Classes/LFDNoise0.schelp [hunk 1]
```diff
@@ -30,32 +30,32 @@ classmethods::
...
-// silent for 2 secs before going up in freq
+// Silence is heard for 2 seconds before any sound is played
 
 { LFNoise0.ar(XLine.kr(0.5, 10000, 3), 0.1) }.scope;
 
 { LFDNoise0.ar(XLine.kr(0.5, 10000, 3), 0.1) }.scope;
 
 
-// LFNoise quantizes time steps at high freqs, LFDNoise does not:
+// LFDNoise does not quantize time steps in high frequencies, in contrary to LFNoise:
```
Comment (telephon) [inline HelpSource/Classes/LFDNoise0.schelp]: better:

// LFDNoise0 does not quantize time steps at high frequencies, in contrast to LFNoise0:

File: HelpSource/Classes/LFDNoise0.schelp [hunk 2]
```diff
@@ -30,32 +30,32 @@ classmethods::
...
+// LFNoise seems stuck, LFDNoise changes on a smooth way.
 
-{ LFNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1)  }.play
+{ LFNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1) }.play
 
-{ LFDNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1)  }.play
+{ LFDNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1) }.play
 
-// silent for 2 secs before going up in freq
+// Silence is heard for 2 seconds before any sound is played
```
Comment (telephon) [inline HelpSource/Classes/LFDNoise0.schelp]: ->
// Silence for 2 seconds before any sound becomes audible

File: HelpSource/Classes/LFDNoise0.schelp [hunk 3]
```diff
@@ -30,32 +30,32 @@ classmethods::
...
+This value is added to the output.
 
 Examples::
 
-code::
-// try wiggling mouse quickly;
-// LFNoise frequently seems stuck, LFDNoise changes smoothly.
+Code::
+// Wiggle mouse in a fast pace;
+// LFNoise seems stuck, LFDNoise changes on a smooth way.
```
Comment (telephon) [inline HelpSource/Classes/LFDNoise0.schelp]: better:
// LFNoise seems stuck, LFDNoise changes smoothly.

File: HelpSource/Classes/LFDNoise0.schelp [hunk 4]
```diff
@@ -30,32 +30,32 @@ classmethods::
 method::ar, kr
 
 argument::freq
-Approximate rate at which to generate random values.
+Random values output rate.
```
Comment (telephon) [inline HelpSource/Classes/LFDNoise0.schelp]: better:
Rate at which random values are generated.

**Comment by tiagmoraismorgado** [comment]
@telephon, you already changed files, right?
cheers. tiago

File: HelpSource/Classes/LFDNoise0.schelp [hunk 5]
```diff
@@ -30,7 +30,7 @@ classmethods::
 method::ar, kr
 
 argument::freq
-Approximate Rate at which random values are generated.
+Approximate random values output rate.
```
Comment (danstowell) [inline HelpSource/Classes/LFDNoise0.schelp]: I'm sorry, this is not as good english as the original.

**Comment by tiagmoraismorgado** [comment]
hey guys i turn out to have reviewed these issues. if you feel like doing so, mege this with the main repository

**Comment by telephon** [comment]
@tiagmoraismorgado thanks! what about the other comments? You have changed only one.

File: HelpSource/Classes/LFDNoise0.schelp (hunk 4 above)
Comment (danstowell) [inline HelpSource/Classes/LFDNoise0.schelp]: No, sorry, that's poorer english than the original

File: HelpSource/Classes/LFDNoise0.schelp [hunk 6]
```diff
@@ -30,33 +30,33 @@ classmethods::
...
-This value will be added to the output.
+This value is added to the output.
 
 Examples::
 
-code::
-// try wiggling mouse quickly;
-// LFNoise frequently seems stuck, LFDNoise changes smoothly.
+Code::
+// Wiggle mouse in a fast pace;
```
Comment (danstowell) [inline HelpSource/Classes/LFDNoise0.schelp]: "in a fast pace" is incorrect english (should be "at" not "in") - really there's no reason to change this line

**Comment by danstowell** [comment]
@tiagmoraismorgado Sorry to be negative but: I must say there are multiple english-language errors in this PR, as there have been in the others. I'm grateful for your contribution but I would encourage you to focus on other aspects and not on editing the english phrasing.

**Comment by tiagmoraismorgado** [comment]
hey dan. i'vê gone throughout. i am aware i am not a nativa english speaker. that doesn't mean i cannot contribute on that aspect. and we are here to learn from the feedback we receive from each other. currently i have to set up my machine from scratch, because my motherboard, charger, and battery died, i had to replace them, and as a matter of fact i am running almost a brand new machine, except that most of the parts are used. and i have to setup my machine again, because the startup disk was formatei by some stupid guys of a store where i took computer, on the forthcoming days i will be focusing mostly on language correction, counting on your help if you don't mind. in mean time, i will start contributing with other things, as i planned (UI, i am now using qt/qml on a daily basis, at work so i can use it to try to improve some stuff on supercollider at that level, on the long run; as well as contributing with source code for the exemples, as i did in the past, and extensions for the supercollider quarks). looking forward to hear from you guys. even if i am not a native speaker, there's always something i can fix, and if i can't fix it 100% accurately, i am here to learn from feedback

**Comment by tiagmoraismorgado** [comment]
by now, i tried to fix all your suggestions. and since supercollider is a collaborative and open-source development thing, nothing better then asking for feedback. i will come back to writing examples for sc documentation, alongside with the process of writing new code and examples, etc.

File: HelpSource/Classes/LFDNoise0.schelp [hunk 7]
```diff
@@ -30,33 +29,33 @@ classmethods::
...
+// LFNoise seems stuck, LFDNoise changes smoothly.
 
-{ LFNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1)  }.play
+{ LFNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1) }.play
 
-{ LFDNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1)  }.play
+{ LFDNoise0.ar(MouseX.kr(0.1, 1000, 1), 0.1) }.play
 
-// silent for 2 secs before going up in freq
+// Silence for 2 seconds before any sound becomes audible
```
Comment (vivid-synth) [inline HelpSource/Classes/LFDNoise0.schelp]: This also changes the meaning of the sentence. Not sure if it helps.

**Comment by jamshark70** [comment]
> on the forthcoming days i will be focusing mostly on language correction, counting on your help if you don't mind.

There are a couple of conflicting concerns here.

- **Plus** - Community participation is *great* -- the community is stronger when users have a stake in SC's progress. The fact that you want to contribute is important and very welcome.

- **Minus** - Pull requests are not free. It costs developer time to decide what to do with them, fix issues etc.

Most of the time, the cost is fairly small while the benefit - to SC's codebase and to community involvement - is much larger. So everybody wins.

This particular set of helpfile pull requests, however - they either need a lot of work or they might just be closed. The cost is still there, but the benefit to the codebase is very small, or zero, so it ends up at best being a wash.

If it's a small number of PRs, the cost remains low. But it seems that you're saying you want to submit a *lot* more such PRs -- so the total cost will keep going up. The cost is worth it if there's a reasonably high chance of them being accepted, but so far, you have to admit that the track record is not quite good.

 Further, the real problems with the help files are:

- Undocumented classes and methods;
- Incomplete documentation of method arguments and returns - expected data types, value restrictions, etc.
- Vague description of behaviors -- e.g., which Collection methods return new collections, and which operate "in place"? Based on the help, you can only guess.

These are way, *way* bigger problems than little tweaks to the wording.

I would strongly suggest that your current approach to helpfile updates is not a wise use of time. I'd suggest the above method-help cleanups as a more productive direction. That is, now you're taking something that is basically ok and making it worse. Better to work on something that is either completely missing, or demonstrably bad (as outlined in those bullets).

> in mean time, i will start contributing with other things, as i planned (UI, i am now using qt/qml on a daily basis, at work so i can use it to try to improve some stuff on supercollider at that level, on the long run; as well as contributing with source code for the exemples, as i did in the past, and extensions for the supercollider quarks).

Jakob had started on some ideas for QML guis but it never really took off. If you could contribute here, that would be **fantastic**!!

**Comment by jamshark70** [comment]
To sum up, my point is not, "Don't contribute" -- it's that there are other ways to contribute that would be more helpful. In particular, the QML idea would be extremely valuable. I'd love to see what you can do with that!

**Comment by jamshark70** [comment]
Oh, one more gap in the documentation: We have almost nothing about how-tos, common usage patterns, recipes, solutions to common problems. One new helpfile about, say, common synth architectures would be worth more than a hundred little wording adjustments.

**Comment by tiagmoraismorgado** [comment]
i am currently already following some of the suggestions. among them, are a QT/QML opensource UI/UX framework, that i started to prototype yesterday, for using on my own applications, and on supercollider attempt to GUI remodel. you can check it here https://github.com/tiagmoraismorgado/TMM_QML_UI_UX_FRAMEWORK_WIP and you can contributo to it, as well, if you wish to do so. this is still not even pre-alpha, but i am planning to take some serious work into the entire thing. kind regras. tiago

**Comment by tiagmoraismorgado** [comment]
furthermore, i am considering to take into some of the suggestions, that you guys gave me regarding documentation, etc. so, thank you very much for your feedback. only with this is kind of community awareness we can make supercollider a better tool. and i am still pretty fresh within helping you guys in terms of development, but i owe a lot, so thanks

**Comment by vivid-synth** [comment]
Closing this one; looking forward to work on undocumented methods/QML work!

**Pull request closed without merging**